### PR TITLE
trie: aggregated path-based state scheme dev

### PR DESCRIPTION
**Pull request by fynnss**
### Description

aggregated path-based state scheme dev

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 

TrieDB
- [x] AggPathDB
    - [x] Write
    - [x] Read
    - [x] Revert

- [x] TrieDataBase

### CODE REVIEW

File: core/rawdb/schema.go [hunk 1]
```diff
@@ -126,6 +126,10 @@ var (
 	trieNodeStoragePrefix = []byte("O") // trieNodeStoragePrefix + accountHash + hexPath -> trie node
 	stateIDPrefix         = []byte("L") // stateIDPrefix + state root -> state id
 
+	// Agg-Path-based storage scheme of merkle patricia trie.
+	aggTrieNodeAccountPrefix = []byte("m") // aggTrieNodeAccountPrefix + hexPath -> agg trie node
```
Comment (joey0612) [inline core/rawdb/schema.go]: following the trieNodeAccountPrefix(PathDB) specification, uppercase(M and G) letters should be more appropriate

File: trie/triedb/aggpathdb/nodebuffer.go [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+	commitTimeTimer.UpdateSince(start)
+	log.Debug("Persisted pathdb nodes", "nodes", len(b.nodes), "bytes", common.StorageSize(size), "elapsed", common.PrettyDuration(time.Since(start)))
+	b.reset()
+	return nil
+}
+
+// aggregateAndWriteNodes will aggregate the trienode into trie aggnode and persist into the database
+// Note this function will inject all the clean aggNode into the cleanCache
+func aggregateAndWriteNodes(batch ethdb.Batch, nodes map[common.Hash]map[string]*trienode.Node,
+	clean *fastcache.Cache, reader ethdb.KeyValueReader) (total int) {
```
Comment (joey0612) [inline trie/triedb/aggpathdb/nodebuffer.go]: ethdb.KeyValueReade is first or second param, maybe better

**Review by joey0612 — APPROVED** [review]

File: core/rawdb/schema.go (hunk 1 above)
Comment (fynnss) [reply core/rawdb/schema.go]: Fixed

File: trie/triedb/aggpathdb/nodebuffer.go (hunk 2 above)
Comment (fynnss) [reply trie/triedb/aggpathdb/nodebuffer.go]: Sure.

**Pull request closed without merging**